### PR TITLE
cleanup(pubsub): move from handwritten topic admin to generated in integration tests

### PR DESCRIPTION
**Pull request by alevenberg**
Part of https://github.com/googleapis/google-cloud-cpp/issues/12987

This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13679)

### CODE REVIEW

File: google/cloud/pubsub/integration_tests/subscription_admin_integration_test.cc [hunk 1]
```diff
@@ -34,6 +34,9 @@ namespace pubsub {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace {
 
+using ::google::cloud::pubsub::Topic;
```
Comment (dbolduc) [inline google/cloud/pubsub/integration_tests/subscription_admin_integration_test.cc:37]: nit: this is superfluous when we are already in the `google::cloud::pubsub` namespace.

**Review by dbolduc — APPROVED** [review]
Consider removing the admin integration tests. What do they buy us?

We are still running the pubsub admin samples, and we have integration tests for the generated spanner admin APIs. Those give us confidence in the generated code.

File: google/cloud/pubsub/integration_tests/topic_admin_integration_test.cc [hunk 2]
```diff
@@ -31,6 +34,9 @@ namespace pubsub {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace {
 
+using ::google::cloud::pubsub::Topic;
```
Comment (alevenberg) [inline google/cloud/pubsub/integration_tests/topic_admin_integration_test.cc:37]:
```suggestion
```

File: google/cloud/pubsub/integration_tests/subscription_admin_integration_test.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/integration_tests/subscription_admin_integration_test.cc:37]:
```suggestion
```

**Comment by alevenberg** [comment]
After moving the code, I'm less inclined to removing them. I will consider

**Review by dbolduc — CHANGES_REQUESTED** [review]
Now that I think about it some more, I don't think we should be changing the integration tests for handwritten clients. Yes the clients are deprecated, but we should probably still have tests that verify that they work.

I think you should only use the generated admin clients in code that is not under test. e.g. if we need to use a TopicAdminClient to create a topic for the SubscriptionAdminClient integration tests.

**Comment by alevenberg** [comment]
> Now that I think about it some more, I don't think we should be changing the integration tests for handwritten clients. Yes the clients are deprecated, but we should probably still have tests that verify that they work.
> 
> I think you should only use the generated admin clients in code that is not under test. e.g. if we need to use a TopicAdminClient to create a topic for the SubscriptionAdminClient integration tests.

I need to change this to add the [[deprecated]] tags

**Comment by alevenberg** [comment]
https://github.com/googleapis/google-cloud-cpp/pull/13677

**Comment by dbolduc** [comment]
I need to change this to add the [[deprecated]] tags

There is a way to suppress the deprecation warnings from breaking our builds, e.g.:

https://github.com/googleapis/google-cloud-cpp/blob/8f7b6654c64cbeb623fdfc2f8bbfd82fdabd25ef/google/cloud/spanner/mocks/row.h#L30-L31

**Comment by alevenberg** [comment]
> > I need to change this to add the [[deprecated]] tags
> 
> There is a way to suppress the deprecation warnings from breaking our builds, e.g.:
> 
> https://github.com/googleapis/google-cloud-cpp/blob/8f7b6654c64cbeb623fdfc2f8bbfd82fdabd25ef/google/cloud/spanner/mocks/row.h#L30-L31

ack

**Comment by alevenberg** [comment]
I disagree though. I think we should not use the old client in the tests. When people see that this code exists, they will use it, and it will make deprecating harder.

**Comment by alevenberg** [comment]
These files should live on past deprecation, so I am moving them to the generated client. I reverted the sub_admin and topic_admin integration tests so we are still testing handwritten code

**Review by dbolduc — APPROVED** [review]

**Pull request merged**